### PR TITLE
core: fix stuffs

### PR DESCRIPTION
**Pull request by missionliao**
Fix issues found by @wmin0 and @Jsying 
- for loop copied data, refer to https://play.golang.org/p/Zy2d60QZlb3
- remove unused functions:
   - types.NewBlock
   - types.RecycleBlock
- avoid init DKG when previous one is not finished.
- reduce log size.

### CODE REVIEW

File: core/consensus.go [hunk 1]
```diff
@@ -1015,8 +1015,11 @@ func (con *Consensus) pullRandomness() {
 		case <-time.After(1500 * time.Millisecond):
 			// TODO(jimmy): pulling period should be related to lambdaBA.
 			hashes := con.ccModule.pendingBlocksWithoutRandomness()
-			con.logger.Debug("Calling Network.PullRandomness", "blocks", hashes)
-			con.network.PullRandomness(hashes)
+			if len(hashes) == 0 {
```
Comment (jimmyhu-dexon) [inline core/consensus.go]: I think you mean `!=0`

File: core/blockpool.go [hunk 2]
```diff
@@ -29,8 +29,8 @@ type blockPool []types.ByPosition
 
 func newBlockPool(chainNum uint32) (pool blockPool) {
 	pool = make(blockPool, chainNum)
-	for _, p := range pool {
-		heap.Init(&p)
+	for i := range pool {
+		heap.Init(&pool[i])
```
Comment (aitjcize) [inline core/blockpool.go]: lol...

Comment (aitjcize) [reply core/blockpool.go]: and this did fail any tests?

Comment (aitjcize) [reply core/blockpool.go]: nvm, because the blockpool is initially empty

**Pull request merged**